lib/Models/Terria.ts

```diff
@@ -1206,34 +1197,32 @@ function generateInitializationUrl(
 }
 
 const loadInitSource = createTransformer(
-  (initSource: InitSource): Promise<JsonObject | undefined> => {
-    let promise: Promise<JsonValue | undefined>;
-
+  async (initSource: InitSource): Promise<JsonObject | undefined> => {
+    let jsonValue: JsonValue | undefined;
     if (isInitUrl(initSource)) {
-      promise = loadJson5(initSource.initUrl);
+      jsonValue = await loadJson5(initSource.initUrl);
     } else if (isInitOptions(initSource)) {
-      promise = initSource.options.reduce((previousOptionPromise, option) => {
-        return previousOptionPromise
-          .then(json => {
-            if (json === undefined) {
-              return loadInitSource(option);
-            }
-            return json;
-          })
-          .catch(_ => {
+      await initSource.options.reduce(async (previousOptionPromise, option) => {
+        try {
+          const json = await previousOptionPromise;
+          if (json === undefined) {
             return loadInitSource(option);
-          });
+          }
+          return json;
+        } catch (_) {
+          return loadInitSource(option);
+        }
       }, Promise.resolve<JsonObject | undefined>(undefined));
```

Comment:
```suggestion
      let error: any;
      for (const option of initSource.options) {
        try {
          jsonValue = await loadInitSource(option);
          if (jsonValue !== undefined) break;
        } catch (err) {
          error = err;
        }
      }
      if (jsonValue === undefined && error !== undefined) throw error;
```

Reply:
I think that is more clear. It was a bit chaotic. The `error = err` but is still a bit weird. You could replace that with a true `TerriaError` with a title and message. We probably will at some point.